src/io/flutter/server/vmService/frame/DartVmServiceValue.java

```diff
@@ -149,10 +149,44 @@ public void computePresentation(@NotNull final XValueNode node, @NotNull final X
     if (computeRegExpPresentation(node)) return;
     if (computeMapPresentation(node)) return;
     if (computeListPresentation(node)) return;
-    computeDefaultPresentation(node);
+
+    // computeDefaultPresentation is called internally here when no result is got.
+    // The reason for this is that the async method used cannot be properly waited.
+    computeDebugPresentation(node);
+
     // todo handle other special kinds: Type, TypeParameter, Pattern, may be some others as well
   }
 
+  private void computeDebugPresentation(final XValueNode node) {
+    myDebugProcess.getVmServiceWrapper()
+      .evaluateInTargetContext(myIsolateId, myInstanceRef.getId(), "toStringDeep()", new VmServiceConsumers.EvaluateConsumerWrapper() {
```

Comment:
Add a TODO here to get the actual DiagnosticsNode objects instead of calling toStringDeep.

Reply:
Added. Please take a look at the latest version.
Just curious. How big difference it will make to get the actual object?
I know you mentioned to me about the performance thing here, but I wonder how big is that and how useful is that compared to allowing people to implement a simple toStringDeep method.

Reply:
Getting the actual objects also lets you provide custom rendering of common property types. For example, based on the structured object view, the inspector window in intellij displays colors like colors and icons like icons.
Additionally, the structured view lets you expand and collapse subtrees for cases where an element has a large # of children.

Reply:
I see. Thanks!
Though it may be in another location of the code.